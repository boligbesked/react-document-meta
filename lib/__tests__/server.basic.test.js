/**
 * @jest-environment node
 */

import React from 'react';
import ReactDOM from 'react-dom/server';
import DocumentMeta from '../server';

DocumentMeta.canUseDOM = false;

describe('DocumentMeta - Server side rendering', () => {
  const DOC_META = {
    title: 'This is a document title',
    description: 'This meta value is describing the page we are looking at',
    canonical: 'http://domain.tld/path/to/page',
    base: 'http://domain.tld',
    meta: {
      charset: 'utf-8',
      name: {
        keywords: 'react,document,meta,tags'
      }
    },
    link: {
      rel: {
        stylesheet: [
          'http://domain.tld/css/vendor.css',
          'http://domain.tld/css/styles.css'
        ]
      }
    }
  };

  it('render document.title / <title> according to the title-attr', () => {
    ReactDOM.renderToString(<DocumentMeta {...DOC_META} />);
    expect(
      DocumentMeta.renderAsHTML().replace(/></g, '>\n<')
    ).toMatchSnapshot();
  });
});
