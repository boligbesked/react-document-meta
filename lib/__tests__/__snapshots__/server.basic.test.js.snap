// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentMeta - Server side rendering render document.title / <title> according to the title-attr 1`] = `
"<title>This is a document title</title>
<base href=\\"http://domain.tld\\" data-rdm/>
<meta charset=\\"utf-8\\" data-rdm/>
<meta name=\\"keywords\\" content=\\"react,document,meta,tags\\" data-rdm/>
<meta name=\\"description\\" content=\\"This meta value is describing the page we are looking at\\" data-rdm/>
<link rel=\\"stylesheet\\" href=\\"http://domain.tld/css/vendor.css\\" data-rdm/>
<link rel=\\"stylesheet\\" href=\\"http://domain.tld/css/styles.css\\" data-rdm/>
<link rel=\\"canonical\\" href=\\"http://domain.tld/path/to/page\\" data-rdm/>"
`;
